tests/backup/backup_test.go

```diff
@@ -549,6 +549,129 @@ var _ = Describe("{BasicBackupCreation}", func() {
+			}
+		}
+	})
+	It("Selective Restore From A Basic Backup", func() {
+
+		Step("Validating deployed applications", func() {
+			log.InfoD("Validating deployed applications")
+			ValidateApplications(contexts)
+		})
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+				bkpLocationName = fmt.Sprintf("%s-%s-bl", provider, getGlobalBucketName(provider))
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				err := CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, "Creating backup location")
+			}
+		})
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying backup cluster with uid: [%s]", clusterUid))
+		})
+		Step("Taking backup of multiple namespaces", func() {
+			log.InfoD("Taking backup of multiple namespaces")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting context")
+			backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, "app-list", time.Now().Unix())
+			err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, bkpNamespaces,
+				labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation with uid: [%s]", clusterUid))
+		})
+		Step("Selecting random backed-up apps and restoring them", func() {
+			log.InfoD("Selecting random backed-up apps and restoring them")
+			selectedBkpNamespaces, err := GetSubsetSlice(bkpNamespaces, len(bkpNamespaces)/2)
+			log.FailOnError(err, "Getting a subset of backed-up namespaces")
+			log.InfoD("Selected application namespaces to restore: %#v", selectedBkpNamespaces)
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			executionMode := Parallel
+			err = TaskHandler(selectedBkpNamespaces, func(namespace string) {
+				restoreName = fmt.Sprintf("%s-%s", "test-restore", namespace)
+				log.InfoD(fmt.Sprintf("Restoring backup [%s] with name - [%s]", namespace, restoreName))
+				err = CreateRestore(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx, make(map[string]string))
```

Comment:
Add timestamp for restore name

Reply:
Added timestamp for both backupName and restoreName.